### PR TITLE
fire: Document flammable group and remove duplicate code

### PR DESCRIPTION
**Pull request by HybridDog**
Furthermore, remove unused abm parameters and reduce indentation

### CODE REVIEW

File: mods/fire/init.lua [hunk 1]
```diff
@@ -107,7 +62,31 @@ minetest.register_node("fire:permanent_flame", {
...
 
 	on_flood = flood_flame,
-})
+}
+
+minetest.register_node("fire:permanent_flame", table.copy(flamedef))
+
+flamedef.description = "Basic Flame"
+flamedef.groups = {igniter = 2, dig_immediate = 3,
+	not_in_creative_inventory = 1}
```
Comment (DS-Minetest) [inline mods/fire/init.lua]: Why not `flamedef.groups.not_in_creative_inventory = 1`?

File: mods/fire/init.lua [hunk 2]
```diff
@@ -339,15 +318,16 @@ if fire_enabled then
 		catch_up = false,
 		action = function(pos, node, active_object_count, active_object_count_wider)
```
Comment (DS-Minetest) [inline mods/fire/init.lua]: Do the same here as in old line 323.

**Comment by paramat** [comment]
:-1: for the de-duplication of flame node registrations, it may be fewer lines but makes the code more complex, less clear, less readable. I'd rather have more lines than a headache.
Everything else may be ok, will review.

File: mods/fire/init.lua (hunk 2 above)
Comment (paramat) [reply mods/fire/init.lua]: Agreed.

**Comment by paramat** [comment]
+1 without nodedef de-duplication and once line comment attended too.

File: game_api.txt [hunk 3]
```diff
@@ -362,6 +362,7 @@ The farming API allows you to easily register plants and hoes.
 Fire API
 --------
 
+Add group flammable when registering a node to make fire seek for it.
```
Comment (Ferk) [inline game_api.txt]: Group flammable can also be applied on items, maybe that should be documented here.
```suggestion
Add group flammable when registering a node to make fire seek for it.
Add it to an item to make it burn up when dropped in lava or fire.
```

Comment (HybridDog) [reply game_api.txt]: I think lava doesn't burn items by default, but there are mods for this.

Comment (Ferk) [reply game_api.txt]: It does: https://github.com/minetest/minetest_game/blob/master/mods/default/item_entity.lua#L56

Comment (HybridDog) [reply game_api.txt]: You're right, sorry.

File: mods/fire/init.lua (hunk 2 above)
Comment (HybridDog) [reply mods/fire/init.lua]: done

**Comment by HybridDog** [comment]
I removed the de-duplication.

**Comment by paramat** [comment]
Thanks, no objection from me. Code looks good but i might do a quick test.

**Comment by paramat** [comment]
Tested :+1:

**Pull request merged**